dapr_agents/observability/instrumentor.py

```diff
@@ -0,0 +1,516 @@
+"""
+OpenTelemetry instrumentor for Dapr Agents with comprehensive tracing and Phoenix UI compatibility.
+
+This instrumentor provides complete observability for Dapr Agents by creating proper
+hierarchical span structures that follow OpenInference semantic conventions. It supports
+all agent types with W3C Trace Context propagation across Dapr Workflow boundaries.
+
+Span Hierarchy by Agent Type:
+
+**Regular Agent (Direct Execution):**
+- Agent.run (AGENT span) - Root span for agent execution
+  └── Agent.process_iterations (CHAIN span) - Processing and reasoning logic
+      ├── Agent.execute_tools (TOOL span) - Batch tool coordination
+      │   └── AgentToolExecutor.run_tool (TOOL span) - Individual tool execution
+      └── ChatClient.generate (LLM span) - Language model interactions
+
+**DurableAgent (Workflow-based with Monitoring):**
+- WorkflowApp.run_and_monitor_workflow_async (AGENT span) - DurableAgent with lifecycle monitoring
+  └── DurableAgent.tool_calling_workflow (CHAIN span) - Workflow orchestration logic
+      ├── WorkflowTask.__call__ (TOOL span) - Tool execution via Dapr Workflow activities
+      └── WorkflowTask.__call__ (LLM span) - LLM calls via Dapr Workflow activities
+
+**Orchestrators (Workflow-based, Fire-and-Forget):**
+- WorkflowApp.run_workflow (AGENT span) - Orchestrator execution without monitoring
+  └── Orchestrator.main_workflow (CHAIN span) - Multi-agent coordination workflow
+      ├── WorkflowTask.__call__ (TASK span) - Workflow orchestration activities
+      └── WorkflowTask.__call__ (LLM span) - LLM calls via Dapr Workflow activities
```

Comment:
hmm so durable agent and orchestrators start with WorkflowApp? Is that bc of the class setups? Is there a way to deliniate which orchestrator based on the span syntax at all?
I suppose a follow up too would be maybe if we can unify some of these spans across agent and durableagent :)

Reply:
Yes, Durable Agent and Orchestrator have (AgenticWorkflow > WorkflowApp) classes underneath. However, they both (by default) start up in a different way. That could be unified in a separate PR and we can adjust the Otel wrappers ;) .